org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -1061,24 +1063,46 @@ else if (m_group.isFunctionColumn(pdc.getAD_PrintFormatItem_ID(), functions[f]))
 								{
 									/** START DEVCOFFEE: script column **/
 									int displayType = pdc.getDisplayType();
-									if (display.startsWith("@SCRIPT")) {
-										display = display.replace("@SCRIPT", "");
-										Object value = parseVariable(display, pdc, pd);
-										Interpreter bsh = new Interpreter ();
-										try {
-											value = bsh.eval(value.toString());
+									MPrintFormatItem pfItem = new MPrintFormatItem(Env.getCtx(), pdc.getAD_PrintFormatItem_ID(), null);
+									//if(rs.getMetaData().getColumnName(displayIndex).contains("SCRIPTCOLUMN")) {	// ScriptColumn
+									if(MPrintFormatItem.PRINTFORMATTYPE_Script.equalsIgnoreCase(pfItem.getPrintFormatType())) {	// ScriptColumn	
```

Comment:
It is more efficient to add PrintFormatType to pdc (PrintDataColumn) and populate that in the DataEngine.getPrintDataInfo method.